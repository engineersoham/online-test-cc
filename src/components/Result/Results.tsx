// import { PieChart } from 'react-minimal-pie-chart';
import { Button } from '@mui/material';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { initialValue } from '../../reducer/reducer';


const Results = () => {
  const RIGHTANSWER = useSelector<initialValue, initialValue['rightans']>(state => state.rightans)
  const WRONGANSWER = useSelector<initialValue, initialValue['wrongans']>(state => state.wrongans)
  const scores = useSelector<initialValue, initialValue['score']>(state => state.score)
  const navigate = useNavigate()

  const start = scores * 60

  const mystyle = {
    height: '20rem',
    width: '20rem',
    backgroundColor: "red",
    backgroundImage: `conic-gradient(green 0deg, green ${start}deg,  red ${start}deg, red 360deg)`,
    borderRadius: "50%"

  }

  return (
    <div >
      
      <div style={{ marginLeft: 50, float: 'left' }}>
      <Button variant='contained' sx={{ml:2,mt:2, mb:1}} onClick={()=>{navigate('/enterpage')}}>Retake</Button>
        <h1 style={{ marginBottom: '40px' }}>Your score is <b>{scores}/6</b></h1>

        <h3><u>RIGHT_ANSWERS</u></h3>
      <ol style={{listStyleType: 'none'}}>
       {RIGHTANSWER.map((item:any) => {
          return (
            <li style={{ color: 'green', marginBottom: '10px' }}> <span style={{color:'black', marginRight:'10px'}}>Q{item.ID}. {item.QUE} </span> <br /> &rarr;  "{item.ANS}"  </li>
          )
        })}
       </ol>

        <h3 style={{ marginTop: '40px' }}><u>WRONG_ANSWERS</u></h3>
        <ol style={{listStyleType: 'none'}}>
        {WRONGANSWER.map((item:any) => {
          return (
            <li style={{ color: 'red', marginBottom: '10px' }}><span style={{color:'black', marginRight:'10px'}}>Q{item.ID}. {item.QUE}  </span><br /> &rarr; "{item.ANS}"  </li>
          )
        })}
        </ol>
      </div>

      <div style={{ float: 'right', marginRight: '10rem', marginTop: '5rem' }}>
      <div title='peichart' style={mystyle}></div>

        {/* <PieChart
          data={[
            { title: 'total', value: 6 - scores, color: 'red', totalValue: 6 },
            { title: 'your score', value: scores, color: 'green', totalValue: 6 }
          ]}
          radius={50}
        />; */}

        

      </div>


    </div>
  )
}

export default Results