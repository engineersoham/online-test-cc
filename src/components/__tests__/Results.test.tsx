import {screen, render} from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../../Store';
import Results from '../Result/Results';

test('snapshot for result page',()=>{
    const res = render(
       <BrowserRouter>
          <Provider store={store}> <Results /></Provider>
       </BrowserRouter>
   )
   expect(res).toMatchSnapshot()
       
})


test('snapshot for result page',()=>{
     render(
        <BrowserRouter>
           <Provider store={store}> <Results /></Provider>
        </BrowserRouter>
    )
    
        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument()
})

test('result style check',()=>{
    render(
        <BrowserRouter>
           <Provider store={store}> <Results /></Provider>
        </BrowserRouter>
    )
        const peichart = screen.getByTitle('peichart');
        expect(peichart).toHaveStyle('borderRadius: 50%')
        expect(peichart).toBeInTheDocument()

})