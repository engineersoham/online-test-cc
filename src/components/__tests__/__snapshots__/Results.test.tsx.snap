// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for result page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div>
        <div
          style="margin-left: 50px; float: left;"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ervl2l-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Retake
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h1
            style="margin-bottom: 40px;"
          >
            Your score is 
            <b>
              0
              /6
            </b>
          </h1>
          <h3>
            <u>
              RIGHT_ANSWERS
            </u>
          </h3>
          <ol
            style="list-style-type: none;"
          />
          <h3
            style="margin-top: 40px;"
          >
            <u>
              WRONG_ANSWERS
            </u>
          </h3>
          <ol
            style="list-style-type: none;"
          />
        </div>
        <div
          style="float: right; margin-right: 10rem; margin-top: 5rem;"
        >
          <div
            style="height: 20rem; width: 20rem; background-color: red; border-radius: 50%;"
            title="peichart"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
     
    <div>
      <div
        style="margin-left: 50px; float: left;"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ervl2l-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Retake
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <h1
          style="margin-bottom: 40px;"
        >
          Your score is 
          <b>
            0
            /6
          </b>
        </h1>
        <h3>
          <u>
            RIGHT_ANSWERS
          </u>
        </h3>
        <ol
          style="list-style-type: none;"
        />
        <h3
          style="margin-top: 40px;"
        >
          <u>
            WRONG_ANSWERS
          </u>
        </h3>
        <ol
          style="list-style-type: none;"
        />
      </div>
      <div
        style="float: right; margin-right: 10rem; margin-top: 5rem;"
      >
        <div
          style="height: 20rem; width: 20rem; background-color: red; border-radius: 50%;"
          title="peichart"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
